Checkbox

Frozenmocca opened the pull request:
Added checkbox functionality; click the checkbox and check appears and strikes out text.

Frozenmocca committed:
Added checkbox functionality

When checkbox is clicked, todo text's h4 class set to strike as styled is associated scss file, and the value of isClicked is passed down to the TodoCheckbox component and the svg visibility of check set from hidden to visible

Frozenmocca committed:
Cleaned Checkbox

Cleaned up comments on Checkbox

Frozenmocca committed:
Merge branch 'develop' into checkbox

Frozenmocca committed:
Merged develop into checkbox

Changed so where strike only occurs when box is clicked, not text

Frozenmocca committed:
Merge branch 'develop' into checkbox

Frozenmocca committed:
Added PropTypes

Frozenmocca committed:
Updated PropTypes

relative-studios commented:
This is almost complete. Although the UI is working properly, you are not actually updating the state of the ```checked property in the original todo state. Also I do like how you implemented the checkmark, I think if you just added a little bit of padding around the check, and make sure that the background color doesn't get updated. You may have to chang how the background color is set for those components, I only wrote it that way initially because I was experimenting with react haha.

relative-studios commented on src/components/todos/TodoItem.js:
```diff
@@ -1,31 +1,49 @@
+        isClicked: !this.state.isClicked
+      });
+    }
+
```
Just out of curiosity, does Treehouse teach you to write class functions inside the render method?

Frozenmocca committed:
Fixed checkbox issues

Changed to update state of checkbox in original todo state. Color of checkbox is now constant, added padding to checkbox for styling

Frozenmocca committed:
Cleaned up one part

Frozenmocca committed:
Slightly changed

Changed slightly so that togglePropertyAt can be used for when the edit component is implimented, making merging easier down the road

relative-studios committed:
Final scaffolding

fml for doing this on this branch, but here is the final scaffolding for the server stuff, the client now talks to the server, and actual todo items get pulled from database

relative-studios committed:
Merge branch 'checkbox' of https://github.com/relative-studios/react-todo into checkbox

relative-studios committed:
Removed console logs and broken method

relative-studios committed:
Set /todos to be a private (logged-in) route

Fixed deleteTodo method
